plugin.rb

```diff
@@ -104,6 +104,20 @@ class Engine < ::Rails::Engine
 
     reaction_users_counting_as_like.flatten!
 
+    # Used to remove historical like data in the filter below.
+    previous_main_reaction_like_ids =
+      object
+        .post_actions
+        .where(
+          "post_actions.post_action_type_id = :like AND
+            post_actions.post_id IN (
+              #{DiscourseReactions::PostActionExtension.post_action_with_reaction_user_sql}
+            )",
+          like: PostActionType.types[:like],
+          valid_reactions: [DiscourseReactions::Reaction.main_reaction_id],
+        )
```

Comment:
Since we are rejecting all `post_actions` that meet this condition eventually, I'm wondering if we can just add this filter to the query below.

```
      object
        .post_actions
        .where(
          "post_actions.post_action_type_id = :like AND
            post_actions.post_id IN (
              #{DiscourseReactions::PostActionExtension.post_action_with_reaction_user_sql}
            )",
          like: PostActionType.types[:like],
          valid_reactions: [DiscourseReactions::Reaction.main_reaction_id],
        )
        .select do |post_action|
         ...
       end
       .reject do |post_action|
        ...
       end
```

Reply:
Oh yeah that might work -- checking.